testhelpers/clock.go

```diff
@@ -19,11 +19,16 @@ func NewFakeSystemClock(n time.Time) *FakeSystemClock {
 }
 
 // Now returns the current value of the FakeSystemClock.
-func (mc *FakeSystemClock) Now() time.Time {
-	return mc.now
+func (fc *FakeSystemClock) Now() time.Time {
+	return fc.now
+}
+
+// Sleep returns immediately.
+func (fc *FakeSystemClock) Sleep(d time.Duration) {
+	return
```

Comment:
This is not a useful implementation. Please follow-up to implement this so that it behaves like a real clock: a call to `Sleep` does not return until the time reported by the clock is `d` greater than the time at which `Sleep` is invoked. This will mean accumulating a list of waiting calls (with a channel or Latch) and checking them all in `Set`.

Reply:
:+1: will merge this with #3355 as a follow on to address this.